### PR TITLE
Modify usage of re.match to fit re.fullmatch or re.search.

### PR DESCRIPTION
**Pull request by gebulmer**
`re.match` features an implicit left anchor, which can be surprising!

in contrast:
`re.fullmatch` features implicit anchors on both sides, but is aptly named
and therefore unsurprising.
`re.search` has no such implicit anchors, which ought to be the default
even if a single anchor is needed.

By modifying the existing instances, we can in future encourage all contributors to use one or the other of the alternatives and ensure that the regex used is the regex that is intended to be used.

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/3382)

### CODE REVIEW

File: qutebrowser/browser/downloads.py [hunk 1]
```diff
@@ -180,7 +180,7 @@ def transform_path(path):
     path = utils.expand_windows_drive(path)
     # Drive dependent working directories are not supported, e.g.
     # E:filename is invalid
-    if re.match(r'[A-Z]:[^\\]', path, re.IGNORECASE):
+    if re.fullmatch(r'[A-Z]:[^\\]', path, re.IGNORECASE):
```
Comment (The-Compiler) [inline qutebrowser/browser/downloads.py]: This should be `re.search` with a `^` anchor added to the regex, as what we want here is really any path starting with something like `E:`.

Comment (gebulmer) [reply qutebrowser/browser/downloads.py]: Noted, corrected :)

**Comment by The-Compiler** [comment]
Hmm, something [seems to be going wrong](https://travis-ci.org/qutebrowser/qutebrowser/jobs/315404445) with PyQt 5.7.1. Can you reproduce this via `tox -e py36-pyqt571 -- tests/end2end/features/test_qutescheme_bdd.py` locally?

**Comment by The-Compiler** [comment]
Looks like Qt 5.7.1 passes now, the other stuff is just some flaky tests... 😟 I'll take a closer look tomorrow!

**Comment by The-Compiler** [comment]
Looks good, thank you!

**Comment by The-Compiler** [comment]
I just noticed a subtle difference when there's a trailing newline:

```
>>> line
'===========\n'
>>> re.match(r'^=+$', line)
<_sre.SRE_Match object; span=(0, 11), match='==========='>
>>> re.fullmatch(r'=+', line)
>>> 
```

I'm quite surprised `re.match` matches here!
This broke building the website in `asciidoc2html.py` because the line wasn't stripped, I'll fix that one, and I *think* nothing else is affected.

**Pull request merged**